rpk: redact secret values when importing from file

r-vasquez opened the pull request:
Now if a value is a secret we will use [redacted] instead of printing the raw value.

```yaml
# Running import when the file have a secret field:
$ rpk cluster config import -f /tmp/config.yaml    
PROPERTY                  PRIOR     NEW
cloud_storage_secret_key  [secret]  [redacted]

Successfully updated configuration. New configuration version is 54.

Cluster needs to be restarted. See more details with 'rpk cluster config status'.

# No edits in rpk cluster config edit:
$ rpk cluster config edit
No changes were made.

# Edit in rpk cluster config edit
$ rpk cluster config edit
PROPERTY                  PRIOR     NEW
cloud_storage_secret_key  [secret]  [redacted]

Successfully updated configuration. New configuration version is 54.

Cluster needs to be restarted. See more details with 'rpk cluster config status'.
```
<!--

See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md##pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED.

Describe, in plain language, the motivation behind the change (bug fix,
feature, improvement) in this PR and how the included commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.

  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.

  Backport of PR #PR-NUMBER

-->

## Backports Required
- [ ] v22.3.x
- [ ] v22.2.x
- [ ] v22.1.x


## Release Notes

  ### Bug Fixes

  * rpk won't print the values of secret properties that are being imported from a file when using `rpk cluster config import/edit`

RafalKorepta reviewed (dismissed):
LGTM:

Could we have some unit tests for the output?
Should this be backported?
I found `v22.1.x`, `v22.2.x` and `v22.3.x` are affected

r-vasquez commented:
Force Push:

- Added ducktape tests
- Added a validation for nil values, so we don't end up changing `<nil>` to `[secret]` when there is no need and can lead to a confusion.

twmb commented on src/go/rpk/pkg/cli/cmd/cluster/config/import.go:
```diff
@@ -104,6 +104,12 @@ func importConfig(
 				}
 			}
 
+			// if the value is [secret] then it means that we didn't change the
+			// secret field value, and it's fine to leave it like that and avoid
+			// printing that we change it from [secret] to [redacted].
```
This comment partially duplicates the comment on line L117 and the first half of it refers to keeping the sentinel value, where the conditional is about changing the value if it isn't a sentinel. wdyt about something like,

```
// If the value is not [secret], the user changed the redacted sentinel
// value and is changing the secret. We redact the value that we store
// to our to-be-printed propertyDeltas.
```

Also, this is redacting the value a bit too early -- the `v` here is saved into `upsert`, meaning the value will be saved as the value `[redacted]`. the modification needs to happen to what is added to `propertyDeltas`, probably in the first two places below.

twmb replied:
Also the TODO below on L120 can probably be deleted -- perhaps there's something about the typing that is different within the `interface{}`, but that's ok.

twmb reviewed (changes requested):
see comment - I think this is bugged so I'm putting a block on it for now to prevent any accidental merge

r-vasquez commented:
/backport v22.2.x

vbotbuildovich commented:
Failed to run cherry-pick command. I executed the below command:

```
git cherry-pick -x f3a40542107e1c0710f91232a29d76137a762b6e
```

[Workflow run logs](https://github.com/redpanda-data/redpanda/actions/runs/3998577807).
